callbacks/preload.go

```diff
@@ -82,27 +83,73 @@ func embeddedValues(embeddedRelations *schema.Relationships) []string {
 	return names
 }
 
-func preloadEmbedded(tx *gorm.DB, relationships *schema.Relationships, s *schema.Schema, preloads map[string][]interface{}, as []interface{}) error {
-	if relationships == nil {
-		return nil
+// preloadEntryPoint enters layer by layer. It will call real preload if it finds the right entry point.
+// If the current relationship is embedded or joined, current query will be ignored.
+//
+//nolint:cyclop
+func preloadEntryPoint(db *gorm.DB, preloadFields, joins []string, relationships *schema.Relationships, preloads map[string][]interface{}, associationsConds []interface{}) error {
+	preloadMap := parsePreloadMap(db.Statement.Schema, preloads)
+
+	// avoid random traversal of the map
+	preloadNames := make([]string, 0, len(preloadMap))
+	for key := range preloadMap {
+		preloadNames = append(preloadNames, key)
+	}
+	sort.Strings(preloadNames)
+
+	joined := func(name string) bool {
+		fullPath := strings.Join(append(preloadFields, name), ".")
+		for _, join := range joins {
```

Comment:
```
DB.Joins("Nested").Joins("Nested.Join").Preload("Nested.Preloads").First(&find).Error

DB.Joins("Nested.Join").Preload("Nested.Preloads").First(&find).Error
```
Both `Preload` and `Joins` will automatically add Nested queries.

Reply:
I fixed it.